### ISSUE TITLE
[nice-2-have] Reference to test case if testcase has failed due to command timeout

### ISSUE THREAD

**Issue by taurus-forever** [aelsabbahy/goss#108]
Hi,

In case if goss.json is huge and complex it is very hard to determine which testcase failed on timeout, like:

```
Failures:
curl -L https://1.1.1.1/api/: exit-status:
Error: Command execution timed out (10s)
curl -L https://1.1.1.1:443/api/: stdout:
Error: Command execution timed out (10s)

Total Duration: 10.015s
Count: 219, Failed: 2
```

In this case it is very hard to determinate which testcase should disabled or tuned to increase timeout until the API performance is improved here, etc. It would be very helpful to show some information here: line number OE expected output OR at least the testcase number in json file, etc.

P.S. we are using template toolkit which generates goss.json for us, so we do not have IP in tt2 hard-coded to use it as reference. Also the same IP is being tested several time using different credentials, transport, ports, etc...

Feel free to resolve it if consider as unnecessary.

Tnx!

**Comment by aelsabbahy** [comment]
So, not 100% sure if the Golang JSON or YAML parsers allow for this to be easily done, that said.. I wonder if "title" and "meta" (arbitrary k/v)  satisfy your needs. [doc](https://github.com/aelsabbahy/goss#patterns-matchers-and-metadata)

Example:
```yaml
command:
  sh -c 'echo "dynamic command"; sleep 10':
    title: Ensuring that backend DB is reachable
    meta:
      test-id: 15
    exit-status: 0
    timeout: 500
```

Output:
```
$ goss validate
F

Failures:

Title: Ensuring that backend DB is reachable
Meta:
    test-id: 15
sh -c 'echo "dynamic command"; sleep 10': exit-status: Error: Command execution timed out (500ms)

Total Duration: 0.501s
Count: 1, Failed: 1
```

**Comment by taurus-forever** [comment]
Yes, "title" and "meta" should help here. I will try them.
Consider as resolved here, I will reopen if necessary, tnx!

**Comment by aelsabbahy** [comment]
FYI, new goss version supports http now, you can try that instead of `curl`.

Example:
```
goss a http https://www.google.com
Adding HTTP to './goss.yaml':

https://www.google.com:
  status: 200
  allow-insecure: false
  timeout: 5000
  body: []
```

Fixed in [v0.1.10](https://github.com/aelsabbahy/goss/releases/tag/v0.1.10)
